Fixes cropping

misaugstad opened the pull request:
This PR should fix two cropping issues:
1. The `pano_yaw_deg` (also called `photographer_heading`) was not being appropriately adjusted like it had been in @tongning 's original code. His original code had the line `pano_yaw_deg = 180 - photographer_heading`. This line was missing in all the code written after him.
1. Similarly, the correction made for issue #2 where we adjust for different resolution images was not being done in any of the code that was written recently.

I've added both of these fixes everywhere that seemed appropriate.

@sarda-devesh could I have you test this code? It might take awhile, but I think every function needs to be tested. I made changes to code that touches just about everything, including the sliding window code. So I would really like to have every single CV tool we have here tested with this new code. And if you don't know how to test one of the CV tools, then we should ask @galenweld to test it (or tell us how to test it).

I don't think there is a rush to get this merged into master, so take your time and we can wait for @galenweld 's help on things. I just really want to make sure I'm not breaking anything before we merge :)

misaugstad committed:
fixes pano_yaw_deg parameter when cropping

misaugstad committed:
cropping now adjusts for image resolution everywhere

sarda-devesh commented:
Hey @misaugstad I think I can through and test every function but I am right now a little busy with school and robotics so I am planning to work on this on the weekend

misaugstad commented:
sure thing, no rush!

sarda-devesh committed:
Finished Base Testing

Fixed issues encountered when running base testing code in samples and still need to do edge case testing

misaugstad commented:
@sarda-devesh it looks like you made a bunch of changes while testing, are these bugs that you found that needed to be fixed? I'd like to know if there are problems with the code I wrote b/c I plan on using it on core Sidewalk as well :grin:
